Marlin/Marlin_main.cpp

```diff
@@ -4295,7 +4295,7 @@ inline void gcode_M109() {
     #define TEMP_CONDITIONS (!residency_start_ms || PENDING(now, residency_start_ms + (TEMP_RESIDENCY_TIME) * 1000UL))
   #else
     // Loop until the temperature is very close target
-    #define TEMP_CONDITIONS (isHeatingHotend(target_extruder))
+    #define TEMP_CONDITIONS (no_wait_for_cooling ? isHeatingHotend(target_extruder) : isCoolingHotend(target_extruder))
```

Comment:
wait for cooling could give unexpected results, e.g.:

```
M109 S210;
...
M109 S25 ; cool down 
```

In summer with 30°C in your workshop it will never be reached.